src/index.ts

```diff
@@ -14,6 +14,601 @@
+   */
+  setMicroseconds(micros: number): string {
+    const abs = Math.abs(micros);
+    let millis = this.getUTCMilliseconds();
+
+    if (abs >= 1000) {
+      millis += Math.floor(abs / 1000) * Math.sign(micros);
+      micros %= 1000;
+    }
+
+    if (Math.sign(micros) === Sign.NEGATIVE) {
+      millis -= 1;
+      micros += 1000;
+    }
+
+    this._micros = micros;
+    this.setUTCMilliseconds(millis);
+
+    return this.getFullTimeString();
+  }
+  /**
+   * Sets the nanoseconds for a specified date according to universal time.
+   *
+   * @param {number} nanoseconds A number representing the microseconds.
+   * @returns {string} Returns a string representing the nanoseconds in the
+   *     specified date according to universal time.
+   *
+   * @example
+   * const date = new PreciseDate();
+   *
+   * date.setNanoseconds(231);
+   *
+   * console.log(date.getNanoseconds());
+   * // expected output: 231
+   */
+  setNanoseconds(nanos: number): string {
+    const abs = Math.abs(nanos);
+    let micros = this._micros;
+
+    if (abs >= 1000) {
+      micros += Math.floor(abs / 1000) * Math.sign(nanos);
+      nanos %= 1000;
+    }
+
+    if (Math.sign(nanos) === Sign.NEGATIVE) {
+      micros -= 1;
+      nanos += 1000;
+    }
+
+    this._nanos = nanos;
+
+    return this.setMicroseconds(micros);
+  }
+  /**
+   * Sets the PreciseDate object to the time represented by a number of
+   * nanoseconds since January 1, 1970, 00:00:00 UTC.
+   *
+   * @param {bigint|number|string} time Value representing the number of
+   *     nanoseconds since January 1, 1970, 00:00:00 UTC.
+   * @returns {string} Returns a string representing the nanoseconds in the
+   *     specified date according to universal time (effectively, the value of
+   *     the argument).
+   *
+   * @see {@link https://github.com/tc39/proposal-bigint|BigInt}
+   *
+   * @example <caption>With a nanosecond string.</caption>
+   * const date = new PreciseDate();
+   * date.setFullTime('1549622069481145231');
+   *
+   * @example <caption>With a BigInt</caption>
+   * date.setFullTime(1549622069481145231n);
+   */
+  setFullTime(time: string|number|bigint): string {
+    if (typeof time !== 'string') {
+      time = time.toString();
+    }
+
+    const sign = Math.sign(Number(time));
+
+    time = time.replace(/^\-/, '');
+
+    const seconds = Number(time.substr(0, time.length - 9)) * sign;
+    const nanos = Number(time.substr(-9)) * sign;
+
+    this.setTime(seconds * 1000);
+    return this.setNanoseconds(nanos);
+  }
+  /**
+   * Sets the PreciseDate object to the time represented by a number of
+   * milliseconds since January 1, 1970, 00:00:00 UTC. Calling this method will
+   * reset both the microseconds and nanoseconds to 0.
+   *
+   * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setTime|Date#setTime}
+   *
+   * @param {number} time Value representing the number of milliseconds since
```

Comment:
Since what? Since what!!!!?

Reply:
Gonna leave this as is so you have to ponder this question for the rest of time

Reply:
And you actually did!

Reply:
This scared me for a seconds, but alas the great mystery has been unveiled - https://github.com/googleapis/nodejs-precise-date/blob/master/src/index.ts#L312